### PR TITLE
fix(openwith): Use fallback UI if an uncaught exception occurs

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

File: src/elements/common/error-boundary/withErrorBoundary.js [hunk 1]
```diff
@@ -5,12 +5,15 @@
...
 import ErrorBoundary from './ErrorBoundary';
 
-const withErrorBoundary = (errorOrigin: ElementOrigin) => (WrappedComponent: React.ComponentType<any>) =>
+const withErrorBoundary = (errorOrigin: ElementOrigin, errorComponent: React.Node = DefaultError) => (
+    WrappedComponent: React.ComponentType<any>,
+) =>
     // $FlowFixMe doesn't know about forwardRef (https://github.com/facebook/flow/issues/6103)
     React.forwardRef((props: Object, ref: React.Ref<any>) => (
-        <ErrorBoundary {...props} errorOrigin={errorOrigin}>
+        <ErrorBoundary {...props} errorOrigin={errorOrigin} errorComponent={errorComponent}>
```
[the remark this answers is not in the document]
Comment (jeremypress) [reply src/elements/common/error-boundary/withErrorBoundary.js]: that sounds reasonable to me. @DanDeMicco any issues from your perspective?

File: src/elements/content-open-with/OpenWithButton.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ type Props = {
     error: ?any,
     icon?: string,
     isLoading: boolean,
-    onClick: Function,
+    onClick?: Function,
```
Comment (jstoffan) [inline src/elements/content-open-with/OpenWithButton.js]: Is a default value required now? It looks like it's being invoked unguarded below.

File: src/elements/common/error-boundary/withErrorBoundary.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanDeMicco) [reply src/elements/common/error-boundary/withErrorBoundary.js]: Should be fine

File: src/elements/content-open-with/OpenWithButton.js (hunk 2 above)
Comment (DanDeMicco) [reply src/elements/content-open-with/OpenWithButton.js]: Can you add an actual function type to this instead  e.g. ```(data:Type) => void```

**Pull request merged**